controller/src/beerocks/master/son_master_thread.cpp

```diff
@@ -3551,6 +3551,33 @@ bool master_thread::handle_cmdu_control_message(const std::string &src_mac,
     }
     case beerocks_message::ACTION_CONTROL_CHANNEL_SCAN_TRIGGER_SCAN_RESPONSE: {
         LOG(TRACE) << "ACTION_CONTROL_CHANNEL_SCAN_TRIGGER_SCAN_RESPONSE for mac " << hostap_mac;
+        auto response =
+            beerocks_header
+                ->addClass<beerocks_message::cACTION_CONTROL_CHANNEL_SCAN_TRIGGER_SCAN_RESPONSE>();
+        if (!response) {
+            LOG(ERROR) << "addClass cACTION_CONTROL_CHANNEL_SCAN_TRIGGER_SCAN_RESPONSE failed";
+            return false;
+        }
+        //get the mac from hostap_mac
+        auto radio_mac = network_utils::mac_from_string(hostap_mac);
+
+        if (!response->success()) {
+            LOG(ERROR) << "failed to trigger scan on radio(" << radio_mac
+                       << "): sending notification to DCS task";
+
+            dynamic_channel_selection_task::sScanEvent new_event;
+            new_event.radio_mac = radio_mac;
+            auto taskId         = database.get_dynamic_channel_selection_task_id(radio_mac);
+            if (taskId == -1) {
+                LOG(ERROR) << "no task for the requested mac (" << radio_mac << ") could be found!";
+                break;
+            }
+
+            tasks.push_event(taskId,
+                             int(dynamic_channel_selection_task::eEvent::SCAN_TRIGGER_FAILED),
```

Comment:
When do we actually fire the `SCAN_TRIGGERED` event?
If `response->success() == true` no event is fired to the task.

Reply:
Just noticed that `SCAN_TRIGGERED` event is fired only after receiving the `ACTION_CONTROL_CHANNEL_SCAN_DUMP_RESULTS_RESPONSE` message.
Is this intentional? Seems a bit odd.

Reply:
the `ACTION_CONTROL_CHANNEL_SCAN_TRIGGERED_NOTIFICATION` is the trigger for sending `SCAN_TRIGGERED` event to the task.
this notification is received from the NL when the scan is actually started.

note that the `ACTION_CONTROL_CHANNEL_SCAN_DUMP_RESULTS_RESPONSE` returns whether the BWL managed to request the scan from the NL - and not the actual scan start.